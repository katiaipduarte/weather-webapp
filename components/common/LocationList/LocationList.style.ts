import styled from 'styled-components'

export const SearchOptions = styled.ul`
	border-bottom: 1px solid white;
	padding-bottom: 2.25rem;
	margin: 0 auto;

	li {
		color: white;
		line-height: 2.5rem;
		cursor: pointer;

		a {
			font-size: 0.75rem;
			font-weight: 400;
		}

		&:hover,
		&:focus,
		&:focus-visible {
			a {
				filter: brightness(75%);
			}
		}
	}
	@media (min-width: 768px) {
		max-width: 32rem;
	}
`
