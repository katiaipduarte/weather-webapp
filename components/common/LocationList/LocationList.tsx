import { GPSLocation } from '@interfaces/open-weather-api/location'
import Link from 'next/link'
import { SearchOptions } from './LocationList.style'

type Props = {
	locations: GPSLocation[]
}

const LocationList = (props: Props): JSX.Element => {
	const { locations } = props

	return (
		<SearchOptions role='list' tabIndex={-1}>
			{locations.length === 0 ? (
				<li data-testid='empty-list'>No results found</li>
			) : (
				<>
					{locations.map((location: GPSLocation) => (
						<li key={location.name} tabIndex={0} role='button'>
							<Link
								href={`/location?lat=${location.lat}&lon=${location.lon}`}
								passHref={true}
							>
								<a data-testid='location-name'>
									{location.name}, {location.country}
								</a>
							</Link>
						</li>
					))}
				</>
			)}
		</SearchOptions>
	)
}

export default LocationList
