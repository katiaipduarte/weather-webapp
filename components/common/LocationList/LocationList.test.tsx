import { mockLocationResponse } from '@mocks/location-response.mock'
import { render } from '@testing-library/react'
import LocationList from './LocationList'

describe('<LocationList /> component Unit Test', () => {
	it('should render 1 <ForecastList /> component', () => {
		const { container } = render(
			<LocationList locations={[mockLocationResponse]} />
		)
		expect(container).toMatchSnapshot()
	})
})
