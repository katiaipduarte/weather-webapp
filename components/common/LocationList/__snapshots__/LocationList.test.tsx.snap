// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationList /> component Unit Test should render 1 <ForecastList /> component 1`] = `
<div>
  <ul
    class="sc-bczRLJ htVIhc"
    role="list"
    tabindex="-1"
  >
    <li
      role="button"
      tabindex="0"
    >
      <a
        data-testid="location-name"
        href="/location?lat=39.74362&lon=-8.80705"
      >
        Leiria
        , 
        PT
      </a>
    </li>
  </ul>
</div>
`;
